pkg/build/controller/build/overrides/overrides.go

```diff
@@ -32,22 +32,28 @@ func (b BuildOverrides) ApplyOverrides(pod *corev1.Pod) error {
 
 	klog.V(4).Infof("Applying overrides to build %s/%s", build.Namespace, build.Name)
 
-	if b.Config.ForcePull {
+	if b.Config.ForcePull != nil {
 		if build.Spec.Strategy.DockerStrategy != nil {
-			klog.V(5).Infof("Setting docker strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
-			build.Spec.Strategy.DockerStrategy.ForcePull = true
+			klog.V(5).Infof("Setting docker strategy ForcePull to %t in build %s/%s", *b.Config.ForcePull, build.Namespace, build.Name)
+			build.Spec.Strategy.DockerStrategy.ForcePull = *b.Config.ForcePull
 		}
 		if build.Spec.Strategy.SourceStrategy != nil {
-			klog.V(5).Infof("Setting source strategy ForcePull to true in build %s/%s", build.Namespace, build.Name)
-			build.Spec.Strategy.SourceStrategy.ForcePull = true
+			klog.V(5).Infof("Setting source strategy ForcePull to %t in build %s/%s", *b.Config.ForcePull, build.Namespace, build.Name)
+			build.Spec.Strategy.SourceStrategy.ForcePull = *b.Config.ForcePull
 		}
 		if build.Spec.Strategy.CustomStrategy != nil {
-			err := applyForcePullToPod(pod)
+			pullPolicy := corev1.PullNever
```

Comment:
This should be `corev1.IfNotPresent`.

Reply:
Updated, thanks.

Reply:
wasn't this resolved incorrectly ... the code change still shows defaulting to Never 

Per https://github.com/kubernetes/api/blob/master/core/v1/types.go#L2243 the default is IfNotPresent anyways

seems like you should leave `applyForcePullToPod` as is and do not add the pullPolicy param